### PR TITLE
http driver

### PR DESCRIPTION
**Pull request by bennyz**
Simple HTTP driver that allows serving files from a directory

### CODE REVIEW

File: contrib/drivers/http/src/jumpstarter_driver_http/client.py [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+        """
+        Upload a file from the local filesystem to the HTTP server.
+
+        Note: This doesn't use HTTP to upload; it streams the file content directly.
+
+        Args:
+            filepath (str): Path to the local file to upload.
+
+        Returns:
+            str: Name of the uploaded file
```
Comment (mangelajo) [inline contrib/drivers/http/src/jumpstarter_driver_http/client.py:65]: Would it make sense to return the URL to access the file?

**Review by mangelajo — COMMENTED** [review]
A very nit suggestion, other than that... NICE!!!

File: contrib/drivers/http/src/jumpstarter_driver_http/driver.py [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+    app: web.Application = field(init=False, default_factory=web.Application)
+    runner: Optional[web.AppRunner] = field(init=False, default=None)
+
+    def __post_init__(self):
+        super().__post_init__()
+        os.makedirs(self.root_dir, exist_ok=True)
+        self.app.router.add_routes([
+            web.get('/{filename}', self.get_file),
+            web.get('/', self.list_files),
+            web.post('/{filename}', self.put_file),
```
Comment (mangelajo) [inline contrib/drivers/http/src/jumpstarter_driver_http/driver.py:55]: post handler takes a request, put_file takes filename + stream.

I think we would need another handler if we want to support that.

File: contrib/drivers/http/src/jumpstarter_driver_http/driver.py [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+    port: int = 8080
+    app: web.Application = field(init=False, default_factory=web.Application)
+    runner: Optional[web.AppRunner] = field(init=False, default=None)
+
+    def __post_init__(self):
+        super().__post_init__()
+        os.makedirs(self.root_dir, exist_ok=True)
+        self.app.router.add_routes([
+            web.get('/{filename}', self.get_file),
+            web.get('/', self.list_files),
```
Comment (mangelajo) [inline contrib/drivers/http/src/jumpstarter_driver_http/driver.py:54]: list_files also takes a request, I believe we need a separate method for the route.

**Review by mangelajo — COMMENTED** [review]
Some extra comments

File: contrib/drivers/http/src/jumpstarter_driver_http/driver.py [hunk 4]
```diff
@@ -0,0 +1,244 @@
...
+        Raises:
+            HttpServerError: If the server fails to stop.
+        """
+        if self.runner is None:
+            logger.warning("HTTP server is not running.")
+            return
+
+        await self.runner.cleanup()
+        logger.info("HTTP server stopped.")
+        self.runner = None
```
[the remark this answers is not in the document]
Comment (bennyz) [reply contrib/drivers/http/src/jumpstarter_driver_http/driver.py:195]: in tftp initially it was started automatically which caused some issues I no longer remember 😅  but I will look into this again

File: contrib/drivers/http/src/jumpstarter_driver_http/driver.py (hunk 2 above)
Comment (bennyz) [reply contrib/drivers/http/src/jumpstarter_driver_http/driver.py:55]: it seems I forgot to push my latest iteration, but I decided to remove all handlers except get_file, the current user of this driver only needs to download a file, the other methods don't need to be served over http

File: contrib/drivers/http/src/jumpstarter_driver_http/driver.py [hunk 5]
```diff
@@ -0,0 +1,244 @@
...
+            if not Path(file_path).resolve().is_relative_to(Path(self.root_dir).resolve()):
+                raise HttpServerError("Invalid target path")
+
+            async with await FileWriteStream.from_path(file_path) as dst:
+                async with self.resource(src_stream) as src:
+                    async for chunk in src:
+                        await dst.send(chunk)
+
+            logger.info(f"File '{filename}' written to '{self.root_dir}'")
+            return filename
```
[the remark this answers is not in the document]
Comment (bennyz) [reply contrib/drivers/http/src/jumpstarter_driver_http/driver.py:91]: yeah I think I somehow messed up and overrode the changes I made in the last force-push https://github.com/jumpstarter-dev/jumpstarter/compare/66ab6d920cd41902b080eaebd2604d232ab274b1..06c727c7f9ea4801c9c73adb1075a828073803ad

😬

**Comment by bennyz** [comment]
ok, hopefully I restored the changes I made, I will avoid force pushing next time 😅

File: contrib/drivers/http/src/jumpstarter_driver_http/driver.py (hunk 3 above)
Comment (bennyz) [reply contrib/drivers/http/src/jumpstarter_driver_http/driver.py]: yeah I did add it and then removed as explained in the other comment

**Comment by mangelajo** [comment]
> ok, hopefully I restored the changes I made, I will avoid force pushing next time 😅

No worries, I force push all the time to keep commits clean.

**Comment by mangelajo** [comment]
Looks good, we need to add it to docs.

**Review by mangelajo — APPROVED** [review]

**Pull request merged**